src/experimental/tools/download2go/types.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright 2019 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IKeySystemOption, IPersistedSessionData, TypedArray } from "../../../core/eme";
+import Manifest, { Representation } from "../../../manifest";
+import { ILocalManifest } from "../../../parsers/manifest/local";
+import { IContextRicher } from "./api/downloader/types";
+
+export type IQualityInputType = "HIGH" | "MEDIUM" | "LOW";
+
+export interface IGlobalSettings {
+  nameDB?: string;
+}
+
+export interface IInitSettings {
+  url: string;
+  transport: "smooth" | "dash";
+  contentID: string;
+  metaData?: {
+    [prop: string]: any;
+  };
+  advanced?: IAdvancedSettings;
+  keySystems?: IKeySystemOption;
+}
+
+export interface IResumeSettings extends IStoredManifest {
+  type: "resume";
+}
+
+export interface IStoredManifest {
+  contentID: string;
+  transport: "smooth" | "dash";
+  manifest: Manifest | null;
+  builder: {
+    video: IContextRicher[];
+    audio: IContextRicher[];
+    text: IContextRicher[];
+  };
+  progress: IProgressBuilder;
+  size: number;
+  metaData?: {
+    [prop: string]: any;
+  };
+  duration: number;
+}
+
+export interface IProgressBuilder {
+  percentage: number;
+  current: number;
```

Comment:
current what?

Reply:
Ha, yes, it's tough to see what it corresponds to at the first see. But If I'd rename the **interface** with a name u mentioned like `IProgressInformation`, would you be ok?

Reply:
Ok never mind, I will rename it.
So the **interface** will be named: **IProgressInformations** as you mentioned.
then, `current` property will be named: `segmentsDownloaded`
`overall` will be named: `totalSegments`